op-chain-ops/cmd/op-upgrade/main.go

```diff
@@ -79,11 +79,11 @@ func entrypoint(ctx *cli.Context) error {
 	}
 
 	superchainName := ctx.String("superchain-target")
-	if superchainName == "" {
-		superchainName, err = upgrades.ToSuperchainName(l1ChainID.Uint64())
-		if err != nil {
-			return err
-		}
+	declaredChainID := superchain.Superchains[superchainName].Config.L1.ChainID
```

Comment:
If the specified `superchain-target` DNE, this is going to panic. Perhaps it'd make more sense to check for existence and gracefully error here?
```suggestion
    specifiedSuperchain, ok := superchain.Superchains[superchainName]
    if !ok {
        return fmt.Errorf("superchain-target %s not registered", superchainName)
    }
	declaredChainID := specifiedSuperchain.Config.L1.ChainID
```

Reply:
Good shout. [7c8bd5d](https://github.com/ethereum-optimism/optimism/pull/9203/commits/7c8bd5da12772f65f40dbe04b1c17ba6e4c7fb09)